cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -232,6 +233,23 @@ def _set_default_instance_role(self):
             ],
         }
 
+        if build.components:
+            for custom_component in build.components:
+                # Check custom component is script, and the url is S3 url
+                if custom_component.type == "script" and custom_component.value.startswith("s3"):
+                    bucket_info = get_bucket_object_name(custom_component.value)
+                    bucket_name = bucket_info.get("bucket_name")
+                    object_name = bucket_info.get("object_name")
+                    action_list = instancerole_policy_document["Statement"][0]["Action"]
+                    if "s3:GetObject" not in action_list:
+                        action_list.append("s3:GetObject")
+                        instancerole_policy_document["Statement"][0]["Action"] = action_list
+                    resources_list = instancerole_policy_document["Statement"][0]["Resource"]
+                    resources_list.append(
+                        core.Fn.sub("arn:${AWS::Partition}:s3:::" + "{0}/{1}".format(bucket_name, object_name))
```

Comment:
`object_name` it's returning only the script name: if the key name contains also a path this will not grant the correct read access. e.g.
```
custom_component.value="s3://test/templates/3.0/post_install.sh"
bucket_name="test"
object_name="post_install.sh"
resource="arn:${AWS::Partition}:s3:::test/post_install.sh"
```
it should be
```
resource="arn:${AWS::Partition}:s3:::test/templates/3.0/post_install.sh" 
```

Reply:
good catch, done